compiler/fir/checkers/src/org/jetbrains/kotlin/fir/analysis/collectors/components/ErrorNodeDiagnosticCollectorComponent.kt

```diff
@@ -35,8 +33,11 @@ class ErrorNodeDiagnosticCollectorComponent(collector: AbstractDiagnosticCollect
     }
 
     override fun visitErrorNamedReference(errorNamedReference: FirErrorNamedReference, data: CheckerContext) {
-        val source = data.qualifiedAccesses.lastOrNull()?.source?.takeIf { it.elementType == KtNodeTypes.DOT_QUALIFIED_EXPRESSION }
-            ?: errorNamedReference.source ?: return
+        val source = data.qualifiedAccesses.lastOrNull()?.takeIf {
+            // Use the source of the enclosing FirQualifiedAccessExpression or FirFunctionCall if it is exactly the call to the erroneous
```

Comment:
`FirFunctionCall` is a subtype of `FirQualifiedAccessExpression`, so I think "the enclosing FirQualifiedAccessExpression if..." seems good enough to me.

I guess your intention was like selector (`.`) or function call, but qualified access includes, e.g., callable reference, so much general notion.

Reply:
OK, I have updated the comment